src/app/services/technical-record/technical-record.service.ts

```diff
@@ -25,8 +39,58 @@ export class TechnicalRecordService {
     return this.store.pipe(select(selectVehicleTechnicalRecordsByVin));
   }
 
-  searchBy(criteria: { type: 'vin'; searchTerm: string }) {
+  searchBy(criteria: { type: SEARCH_TYPES; searchTerm: string }) {
     const { type, searchTerm } = criteria;
-    this.store.dispatch(getByVIN({ [type]: searchTerm }));
+
+    switch (type) {
+      case SEARCH_TYPES.VIN:
+        this.store.dispatch(getByVIN({ [type]: searchTerm }));
+        break;
+      case SEARCH_TYPES.PARTIAL_VIN:
+        this.store.dispatch(getByPartialVIN({ [type]: searchTerm }));
+        break;
+    }
+  }
+
+  /**
+   * A function to get the correct tech record to create the summary display which uses time first then status code
+   * @param record This is a VehicleTechRecordModel passed in from the parent component
+   * @returns returns the tech record of correct hierarchy precedence or if none exists returns undefined
+   */
+  viewableTechRecord(record: VehicleTechRecordModel, destroy$: Subject<any>): TechRecordModel | undefined {
```

Comment:
Couple of observations here...
1. Could this be a completely asynchronous function so there is no race condition between the if on line 71 and the subscription. You could map the params and try to find the tech record, if not found resolve to the status code filter.
2. could al of this be done as a selector since all of this data should be in state?

Reply:
This has now been addressed, thanks for the suggestion.

I will try and turn it into a selector in one of the next PRs as there's a couple more to finish off the search functionality.